scripts/generate_enums.py

```diff
@@ -0,0 +1,356 @@
+            value = enum_value.find('initializer').text.removeprefix('= ')
+            enum_value_doc = parse_doc(enum_value)
+            values.append(EnumValue(enum_name, value, enum_value_doc))
+
+        enum_definitions.append(EnumDefinition(name, values, doc))
+
+    return enum_definitions
+
+
+def parse_doc(elem):
+    items = []
+    brief = elem.find('briefdescription/para')
+    items.append(parse_doc_elem(brief))
+
+    for para in elem.findall('detaileddescription/para'):
+        items.append(parse_doc_elem(para))
+
+    return DocComment(items)
+
+
+def strip_text(text):
+    if text:
+        strip = text.strip()
+        if strip:
+            return strip
+    return None
+
+
+def parse_doc_elem(elem: ElementTree.Element):
+    items = []
+    tag = elem.tag
+    parse_children = True
+    text = strip_text(elem.text)
+    if tag == "para":
+        if text:
+            items.append({"type": "text", "text": text})
+    elif tag == "ref":
+        if text:
+            items.append({"type": "ref", "text": text})
+    elif tag == "computeroutput":
+        if text:
+            items.append({"type": "code", "text": text})
+    elif tag == "itemizedlist":
+        values = []
+        for li in elem.findall("listitem"):
+            li_values = []
+            for e in li:
+                li_values.extend(parse_doc_elem(e))
+            values.append(li_values)
+        items.append({"type": "list", "values": values})
+        parse_children = False
+    else:
+        print(f"unknown tag = {tag}, consider adding it to parse_doc_elem")
+    if parse_children:
+        for item in elem:
+            items.extend(parse_doc_elem(item))
+            if item.tail:
+                strip = item.tail.strip()
+                if strip:
+                    items.append({"type": "text", "text": strip})
+    return items
+
+
+def generate_enums(generator: EnumGenerator, enum_definitions: list[EnumDefinition]):
+    for enum_definition in enum_definitions:
+        generator.generate_enum(enum_definition)
+
+
+def main():
+    os.chdir(os.path.dirname(os.path.realpath(__file__)))
+
+    parser = argparse.ArgumentParser(description='Generate enums')
+
+    parser.add_argument('--type', choices=['all', 'java', 'go'], help='Type of enum generation', required=True)
+    parser.add_argument('--go_output_dir',
+                        default=ROC_GO_BASE_PATH, help=f"Go output directory (default: {ROC_GO_BASE_PATH})")
+    parser.add_argument('--java_output_dir',
+                        default=ROC_JAVA_BASE_PATH, help=f"Java output directory (default: {ROC_JAVA_BASE_PATH})")
+
+    args = parser.parse_args()
+
+    xml_doc = parse_config_xml()
+    enum_definitions = get_enums(xml_doc)
+
+    name_prefixes = {}
+    for enum_definition in enum_definitions:
+        name = enum_definition.name
+        prefix = ODD_PREFIXES.get(name, name.upper() + "_")
+        name_prefixes[name] = prefix
```

Comment:
nit: I'd move this into a function to keep main() high-level

Reply:
done